### DISCUSSION TITLE
Feature columns pre-processing

### DISCUSSION THREAD

**Discussion by carlota-moh** [NannyML/nannyml#166] [Q&A]
First of all, thank you so much for this amazing library! I've been racking my brain on how to perform uncertainty estimation and this is just what I needed

My question is rather simple: does the Nanny estimator perform any preprocessing on the input data? Scanning through the code I have noticed that it does encode categorical variables, but does it scale numeric variables too? If not, does it expect to have them already scaled when feeding data to the model?

**Comment by jakubnml** [comment]
Hi @carlota-moh, can you explicitly name the estimator you have in mind and maybe tell us more about your use case? Just to make sure I will answer the best I can.

**Reply by carlota-moh** [reply]
Yes! I am using DLE to predict the absolute error on a regression problem. For the moment I am just following the documentations's example, but I would like to use it on some other datasets and get the feel of how it works. What I usually tend to do is scale numerical features prior to training and testing the models, so I was wondering if the DLE estimator uses the scaled variables or just the original ones.

**Reply by jakubnml** [reply]
So DLE encodes categorical variables to make them numeric and start from 0 as this is the most convenient for LGBMRegressor it uses. So it does not really change the information that a feature contains. DLE does not do anything with continuous features. I think it is best if you provide the features in the same way you input them for the model you want to monitor with NannyML (already preprocessed).
